lib/ex_doc/formatter/html/autolink.ex

```diff
@@ -87,37 +87,49 @@ defmodule ExDoc.Formatter.HTML.Autolink do
   Converts the given `ast` to string while linking the locals
   given by `typespecs` as HTML.
   """
-  def typespec(ast, typespecs, aliases, lib_dirs \\ elixir_lib_dirs())
+  def typespec(ast, typespecs, aliases, lib_dirs \\ elixir_lib_dirs()) do
+    if formatter_available?() do
+      typespec_to_string(ast, typespecs, aliases, lib_dirs)
+    else
+      format_typespec(ast, typespecs, aliases, lib_dirs)
+    end
+  end
 
-  def typespec({:when, _, [{:::, _, [left, {:|, _, _} = center]}, right]} = ast, typespecs, aliases, lib_dirs) do
+  defp format_typespec({:when, _, [{:::, _, [left, {:|, _, _} = center]}, right]} = ast, typespecs, aliases, lib_dirs) do
     if short_typespec?(ast) do
```

Comment:
I don't think we need this? We should let the formatter handle it all, no?

Reply:
Just to clarify, `format_typespec`, `short_typespec`, `format_typespec_with_new_line`, `normalize_left` will all be gone when we depend on 1.6

Reply:
Oh, sorry. I got confused. I thought `format_typespec`used the formatter (since it says format in the name) and I thought that `typespec_to_string` used `Macro.to_string` since it has `to_string`. Should we flip the names?

Reply:
doh, good call! All fixed now